File: FinanceApp.Web/src/models/ExpenseTransactionDtos/GetExpenseTransactionDto.ts
import { Money } from "../Money/Money";
import { GetExpenseTransactionGroupDto } from "./GetExpenseTransactionGroupDto";

export interface GetExpenseTransactionDto {
    id: string;  
    name: string;
    description?: string;
    value: Money;
    dueDate?: string;  
    priority?: number;
    transactionGroup?: GetExpenseTransactionGroupDto;
}  

File: FinanceApp.Application/Dtos/ExpenseTransactionDtos/GetExpenseTransactionDto.cs
using FinanceApp.Application.Dtos.ExpenseTransactionGroupDtos;
using FinanceApp.Domain.Entities;

namespace FinanceApp.Application.Dtos.ExpenseTransactionDtos;

public class GetExpenseTransactionDto
{
  #region Properties

  public Guid Id { get; set; }

  public string Name { get; set; } = string.Empty;

  public string? Description { get; set; }

  public Money Value { get; set; } = new();

  public DateTimeOffset? DueDate { get; set; }

  public int? Priority { get; set; }

  public GetExpenseTransactionGroupDto? TransactionGroup { get; set; }

  #endregion
}
